test/dynamo/test_skip_non_tensor.py

```diff
@@ -170,6 +170,8 @@ def test_do_not_skip_side_effects(self):
         global _variable, _variable_2
 
         for mode in range(1, 7):
+            torch._dynamo.reset()
```

Comment:
What is the reason for recompilation?

Reply:
Use TORCH_LOGS="recompiles"

Reply:
https://github.com/pytorch/pytorch/issues/127483 Have all the details each iteration we _variable is updated and mode.